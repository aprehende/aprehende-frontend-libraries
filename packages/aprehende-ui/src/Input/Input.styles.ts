import styled from '@emotion/styled';

export const InputContainer = styled.div<{ fullWidth?: boolean }>`
  width: ${({ fullWidth }) => (fullWidth ? '100%' : '250px')};
  box-sizing: border-box;
  display: flex;
  padding: 10px;
  border-radius: 5px;
  border: 1px solid #ccc;
  & > *:not(:last-child) {
    margin-right: 10px;
  }
  & > input {
    margin: 0;
    padding: 0;
    width: 100%;
    border: none;
    outline: none;
    margin-right: 10px;
  }
`;
