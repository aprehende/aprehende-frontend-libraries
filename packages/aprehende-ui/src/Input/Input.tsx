import React from 'react';

import { InputContainer } from './Input.styles';

interface InputProps extends React.InputHTMLAttributes<HTMLInputElement> {
  widthFull?: boolean;
  startRender?: React.ReactNode;
  endRender?: React.ReactNode;
}

const Input = React.forwardRef<HTMLInputElement, InputProps>(
  ({ startRender, endRender, widthFull, ...rest }, ref) => {
    return (
      <InputContainer fullWidth={widthFull}>
        {startRender && <span>{startRender}</span>}
        <input ref={ref} {...rest} />
        {endRender && <span>{endRender}</span>}
      </InputContainer>
    );
  }
);

export default Input;
