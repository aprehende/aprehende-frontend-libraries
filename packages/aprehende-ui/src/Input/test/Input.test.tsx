import { fireEvent, render } from '@testing-library/react';
import { describe, it, vi } from 'vitest';

import Input from '../Input';

describe('Input', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Input placeholder="Email" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('placeholder should be Email', () => {
    const { getByPlaceholderText } = render(<Input placeholder="Email" />);
    expect(getByPlaceholderText('Email')).toBeInTheDocument();
  });

  it('should render startRender', () => {
    const { getByText } = render(
      <Input placeholder="Email" startRender={<span>0/50</span>} />
    );
    expect(getByText('0/50')).toBeInTheDocument();
  });

  it('should render endRender', () => {
    const { getByText } = render(
      <Input placeholder="Email" endRender={<span>0/50</span>} />
    );
    expect(getByText('0/50')).toBeInTheDocument();
  });

  it('onChangeFunc should be called', () => {
    const onChangeFunc = vi.fn();
    const { getByPlaceholderText } = render(
      <Input placeholder="Email" onChange={onChangeFunc} />
    );
    fireEvent.change(getByPlaceholderText('Email'), {
      target: { value: 'test' },
    });
    expect(onChangeFunc).toHaveBeenCalled();
  });

  it('should render with 100% width', () => {
    const { getByPlaceholderText } = render(
      <Input placeholder="Email" widthFull />
    );
    expect(getByPlaceholderText('Email')).toHaveStyle('width: 100%');
  });
});
