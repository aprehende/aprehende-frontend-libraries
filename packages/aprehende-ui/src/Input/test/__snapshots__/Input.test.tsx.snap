// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input > should render correctly 1`] = `
<DocumentFragment>
  <div
    class="css-q5ir41"
  >
    <input
      placeholder="Email"
    />
  </div>
</DocumentFragment>
`;
